components/ingest-service/server/jobs.go

```diff
@@ -1,124 +1,280 @@
 package server
 
 import (
-	"github.com/chef/automate/components/ingest-service/backend"
-	"github.com/chef/automate/components/ingest-service/config"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	log "github.com/sirupsen/logrus"
+	rrule "github.com/teambition/rrule-go"
 	"golang.org/x/net/context"
+
+	"github.com/chef/automate/api/interservice/ingest"
+	"github.com/chef/automate/components/ingest-service/backend"
+	"github.com/chef/automate/components/ingest-service/config"
+	"github.com/chef/automate/lib/cereal"
 )
 
-func (server *JobSchedulerServer) addUpdateJob(jc config.JobConfig) {
-	var cronFunc func()
+const (
+	MissingNodesJobName                 = "missing_nodes"
+	MissingNodesScheduleName            = "periodic_missing_nodes"
+	DeleteNodesJobName                  = "delete_nodes"
+	DeleteNodesScheduleName             = "periodic_delete_nodes"
+	MissingNodesForDeletionJobName      = "missing_nodes_for_deletion"
+	MissingNodesForDeletionScheduleName = "periodic_missing_nodes_for_deletion"
+)
+
+func InitializeJobManager(c *cereal.Manager, client backend.Client) error {
+	err := c.RegisterTaskExecutor(DeleteNodesJobName, &DeleteExpiredMarkedNodesTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	err = c.RegisterTaskExecutor(MissingNodesJobName, &MarkNodesMissingTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	err = c.RegisterTaskExecutor(MissingNodesForDeletionJobName, &MarkMissingNodesForDeletionTask{client}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+	for _, jobName := range []string{MissingNodesJobName, DeleteNodesJobName, MissingNodesForDeletionJobName} {
+		err = c.RegisterWorkflowExecutor(jobName, NewSingleTaskWorkflow(jobName))
+		if err != nil {
+			return errors.Wrapf(err, "failed to register workflow for %q", jobName)
+		}
+	}
+	return nil
+}
+
+func MigrateJobsSchedule(c *cereal.Manager, oldConfigFile string) error {
+	jc, err := config.OldJobConfigFromFile(oldConfigFile)
+	if err != nil {
+		log.WithError(err).Warn("failed to read old job config from disk, defaults will be used")
+	}
+
+	for idx, name := range config.JobList {
+		scheduleName := fmt.Sprintf("periodic_%s", name)
+		config, err := jc.ConfigForJob(idx)
```

Comment:
`jc` could be `nil` here should we still use this variable?

Reply:
ConfigForJob handles the nil case.  It is a bit ugly, and most of the ugliness is stemming from attempting to handle the case of potentially having a partial on-disk config.